test/boost/secondary_index_test.cc

```diff
@@ -42,6 +42,31 @@ SEASTAR_TEST_CASE(test_secondary_index_regular_column_query) {
     });
 }
 
+SEASTAR_TEST_CASE(test_view_builder_use_after_free) {
```

Comment:
I think this test will be very enigmatic to a person who sees it for the first time without any context. In a comment, please mention for which issue this is a regression test for, and explain how it reproduces the issue.

Besides, I don't think that this is a minimal reproducer. You just copy-pasted the test above while many of the actions and the assertions here are not needed. I think it is only sufficient to create a table and a single index, that should reproduce the issue because it triggers the view building logic which was causing problems before the patch.